### ISSUE TITLE
Sandra hand cameras wrong orientation

### ISSUE THREAD

**Issue by osrf-migration** [osrf/drcsim#401]
**Original report ([archived issue](https://osrf-migration.github.io/drcsim-gh-pages/#!/osrf/drcsim/issues/401)) by Anonymous.**

----------------------------------------
Hand camera view should be out wards from palms.  Instead, view is outwards from back of hand

**Comment by osrf-migration** [comment]
**Original comment by Steve Peters (Bitbucket: [Steven Peters](https://bitbucket.org/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D/), GitHub: [scpeters](https://github.com/scpeters)).**

----------------------------------------
Can you provide any information about your setup and some example images that you consider incorrect?

**Comment by osrf-migration** [comment]
**Original comment by Yohei Kakiuchi (Bitbucket: [YoheiKakiuchi](https://bitbucket.org/%7Beaf4691e-d19c-42e6-97bc-76ae3ba2e985%7D/)).**

----------------------------------------
I added example image for this problem.
I just launched 3 programs and added box objects behind Atlas and sphere objects in front of Atlas.
I think hand cameras are expected to see sphere objects in front of Atlas, but in the viewers I can see box objects. I think they are incorrect.
~~~
$ roslaunch drcsim_gazebo atlas_v3_sandia_hands.launch
$ rosrun image_view image_view image:=/sandia_hands/l_hand/camera/left/image_color
$ rosrun image_view image_view image:=/sandia_hands/r_hand/camera/right/image_colo
~~~
![simple test](https://lh5.googleusercontent.com/-c0wnVcAdEaU/UqyGrI5UsjI/AAAAAAAABSA/KjwRHMcJ_Y4/s144/atlas_v3_sandia_hands_hand_images.png)

**Comment by osrf-migration** [comment]
**Original comment by Steve Peters (Bitbucket: [Steven Peters](https://bitbucket.org/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D/), GitHub: [scpeters](https://github.com/scpeters)).**

----------------------------------------
Can you attach larger images? I can't see the Atlas hand orientation very clearly in the image you posted.

**Comment by osrf-migration** [comment]
**Original comment by Yohei Kakiuchi (Bitbucket: [YoheiKakiuchi](https://bitbucket.org/%7Beaf4691e-d19c-42e6-97bc-76ae3ba2e985%7D/)).**

----------------------------------------
![larger image](https://lh5.googleusercontent.com/-c0wnVcAdEaU/UqyGrI5UsjI/AAAAAAAABSA/KjwRHMcJ_Y4/s800/atlas_v3_sandia_hands_hand_images.png)

I added larger image.
This pose is just standing pose, after starting gazebo I didn't send any commands to Atlas.

**Comment by osrf-migration** [comment]
**Original comment by John Hsu (Bitbucket: [hsu](https://bitbucket.org/%7B0a186eae-abf0-4514-a951-23db5eccc286%7D/), GitHub: [hsu](https://github.com/hsu)).**

----------------------------------------
could this be related to [gazebo issue 920](https://bitbucket.org/osrf/gazebo/issue/920/camera-rotation-error)?

**Comment by osrf-migration** [comment]
**Original comment by Steve Peters (Bitbucket: [Steven Peters](https://bitbucket.org/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D/), GitHub: [scpeters](https://github.com/scpeters)).**

----------------------------------------
That's a good point @hsu ; there was a problem with the camera orientation in gazebo 2.0.0 that we fixed in gazebo 2.1.0. Can you confirm which version of gazebo you are using? I believe if you `apt-get update` and install the latest version that it will work properly.

**Comment by osrf-migration** [comment]
**Original comment by Yohei Kakiuchi (Bitbucket: [YoheiKakiuchi](https://bitbucket.org/%7Beaf4691e-d19c-42e6-97bc-76ae3ba2e985%7D/)).**

----------------------------------------
I used current gazebo-current package.
But this problem still occur.
Please check it on your system.

~~~
$ LANG=C aptitude show drcsim
Package: drcsim                          
New: yes
State: installed
Automatically installed: no
Version: 3.1.1-2~precise
Priority: extra
Section: devel
Maintainer: Brian Gerkey <gerkey@osrfoundation.org>
Architecture: amd64
Uncompressed Size: 84.3 M
Depends: gazebo-current, ros-groovy-pr2-mechanism, ros-groovy-std-msgs, ros-groovy-common-msgs, ros-groovy-image-common,
         ros-groovy-geometry, ros-groovy-pr2-controllers, ros-groovy-geometry-experimental, ros-groovy-robot-model-visualization,
         ros-groovy-image-pipeline, ros-groovy-image-transport-plugins, ros-groovy-xacro, ros-groovy-robot-state-publisher,
         ros-groovy-gazebo-ros-current (>= 2.4.0), ros-groovy-gazebo-plugins-current (>= 2.4.0), ros-groovy-gazebo-msgs-current (>=
         2.4.0), osrf-common (>= 1.1.0), sandia-hand (>= 5.2.0)
Conflicts: drcsim-nightly, drcsim-nightly, drcsim-prerelease, drcsim-prerelease, drcsim
Description: Darpa Robotics Challenge (DRC) simulator
 Building upon Gazebo simulator, the DRC Simulator, or DRCSim, comprises additional worlds, models, and code that is specifically
 required to simulate the DRC domain.
Homepage: http://theroboticschallenge.org
~~~

~~~
$ LANG=C aptitude show gazebo-current
Package: gazebo-current                  
New: yes
State: installed
Automatically installed: yes
Version: 2.1.0-1~precise
Priority: extra
Section: devel
Maintainer: Nate Koenig <nate@osrfoundation.org>
Architecture: amd64
Uncompressed Size: 62.1 M
Depends: libfreeimage3, libprotoc7, libprotobuf-dev, freeglut3, libcurl4-openssl-dev, libtinyxml2.6.2, libtar0, libtbb2,
         libtbb-dev, libogre-dev, libxml2, pkg-config, libqtcore4, libqtgui4, libltdl7, libboost-thread1.46.1,
         libboost-signals1.46.1, libboost-system1.46.1, libboost-filesystem1.46.1, libboost-program-options1.46.1,
         libboost-regex1.46.1, libboost-iostreams1.46.1, robot-player, libcegui-mk2-0.7.5, libcegui-mk2-dev, libavformat53 |
         libavformat-extra-53, libavcodec53 | libavcodec-extra-53, libswscale2 | libswscale-extra-2, libsdformat-dev (>= 1.4.7~),
         ttf-dejavu-core
Conflicts: gazebo, gazebo, gazebo-nightly, gazebo-nightly, gazebo-prerelese, gazebo-prerelese, gazebo-current
Provides: gazebo-provider
Description: Gazebo simulator - Main package
 Gazebo is a multi-robot simulator for outdoor environments. Like Stage (part of the Player project), it is capable of simulating a
 population of robots, sensors and objects, but does so in a three-dimensional world. It generates both realistic sensor feedback
 and physically plausible interactions between objects (it includes an accurate simulation of rigid-body physics). 
 
 This package contains all the main files to make the simulator work, including executables, libraries, models, etc.
Homepage: http://gazebo-currentsim.org
~~~

**Comment by osrf-migration** [comment]
**Original comment by Steve Peters (Bitbucket: [Steven Peters](https://bitbucket.org/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D/), GitHub: [scpeters](https://github.com/scpeters)).**

----------------------------------------
I just recently returned to this issue, and I am now able to reproduce it, whereas I was unable to do so before. Currently investigating...

EDIT: I just realized that my previous tests involved atlas_sandia_hands, not atlas_v3_sandia_hands. Sorry for not reading your bug report properly. I'll get to the bottom of this.

**Comment by osrf-migration** [comment]
**Original comment by John Hsu (Bitbucket: [hsu](https://bitbucket.org/%7B0a186eae-abf0-4514-a951-23db5eccc286%7D/), GitHub: [hsu](https://github.com/hsu)).**

----------------------------------------
[This patch should fix it](https://bitbucket.org/osrf/gazebo/commits/298662e1bd0235860b52dc2b135e31fcf43d465c), however, we need an unit test and clean up before making an official pull request.

**Comment by osrf-migration** [comment]
**Original comment by John Hsu (Bitbucket: [hsu](https://bitbucket.org/%7B0a186eae-abf0-4514-a951-23db5eccc286%7D/), GitHub: [hsu](https://github.com/hsu)).**

----------------------------------------
reopening [gazebo issue 920](https://bitbucket.org/osrf/gazebo/issue/920/camera-rotation-error) as there is a regression from gazebo 1.9 to gazebo 2.0.  A test has been added (mentioned in 920) that will catch this error.

**Comment by osrf-migration** [comment]
**Original comment by John Hsu (Bitbucket: [hsu](https://bitbucket.org/%7B0a186eae-abf0-4514-a951-23db5eccc286%7D/), GitHub: [hsu](https://github.com/hsu)).**

----------------------------------------
[gazebo pull request 916](https://bitbucket.org/osrf/gazebo/pull-request/916/fix-for-camera-rotation-bug-issue-920/diff) should fix this issue.

**Comment by osrf-migration** [comment]
**Original comment by Nate Koenig (Bitbucket: [Nathan Koenig](https://bitbucket.org/%7Bc862cdd9-fcc8-4419-9fc7-e20db14b8fcb%7D/)).**

----------------------------------------

* changed state from "new" to "resolved"

Fixed in [pull request #916](https://osrf-migration.github.io/drcsim-gh-pages/#!/osrf/drcsim/pull-requests/916)
